ui-framework-documentation.md

```diff
@@ -228,20 +234,59 @@ sub run {
 }    
 ``` 
 
-But in case if need to verify some negative cases the method cannot be
-used as is. For instance, to verify that prompt appears when blank
-password is entered. For such kind of cases the access to the
-[Pages](#page) through [Controller](#controller) is added.  
-Then [Test Module](#test-module) code may look like:
+
+
+#### Test Data usage in Controller
+
+Do not define any test data in [Controller](#controller) layer as it
+could make test maintenance more complicated. Use the test data passed
+from the [Test Module](#test-module) layer instead.
+
+Example:
+
+
+* To make an action depending on a value, use the data from test data:
```

Comment:
I'm not sure about that, I think it is better to create different methods, those switches might not be best practice but of course can work out. The thing is that the tester is as specific as the business logic he/she is using with that, so if it is chosen `create_encrypted_partition`, the expectation is to have an encrypted partition, if it is lvm or not is not specified, so in this case we could do the switch, but ideally if we expose in the test the method create_lvm_encrypted_partition then the expectation are more clear.

Reply:
I would say, generally yes but it depends, we may prefer this over two almost identical methods. But you're right this part is not clear.

Reply:
changed